Fix provider.yaml errors due to exit(0) in test

potiuk opened the pull request:
There were a few errors in provider structure resulting from
accidental exit(0) in provider's tests. Luckily none of the changes
have been released yet.

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

kaxil commented on airflow/providers/amazon/aws/operators/s3_upload.py:
```diff
@@ -21,7 +21,7 @@
 
 
-class LocalFilesystemToS3Operator(BaseOperator):
+class S3UploadOperator(BaseOperator):
```
hmm.. -- why this change?

potiuk replied:
Because if we have "transfer" we need to have two integration "source" and "target" in provider.yaml (and in the documentation) . The "transfer" operators by definition take two Hooks and perform active transfer operation between them. In this case we have no "Source" integration because we are simply uploading a local file to S3.

potiuk replied:
See here for example: https://github.com/apache/airflow/blob/e18b6a6d19f9ea0d8fe760ba00adf38810f0e510/airflow/providers/amazon/provider.yaml#L408

potiuk replied:
It was simply wrongly classified type of operator. This operator is a "classic" operator not "transfer" one.

potiuk replied:
UPDATE: My bad. `Local` integration is way better here.

potiuk committed:
Fix provider.yaml errors due to exit(0) in test

There were a few errors in provider structure resulting from
accidental exit(0) in provider's tests. Luckily none of the changes
have been released yet.
